### DISCUSSION TITLE
[과목 1] 제 2장 데이터 모델과 SQL 연습문제

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#5] [연습문제] [answered]
### 연습문제

**문제 1. 아래 데이터 모델을 보고 몇 정규형 위배인지 고르시오.**

![스크린샷 2023-04-13 오후 9 52 11](https://user-images.githubusercontent.com/67765871/231764241-83589055-db68-4c6e-b5b6-4b551421ff05.png)

① 제1정규형
② 제2정규형
③ 제3정규형
④ 보이스코드정규형

**문제 2. 다음 중 조인에 대한 설명으로 가장 부적절한 것을 고르시오.** 
① 조인은 엔터티 간 관계에서 비롯된다.
② 조인은 매핑키를 통해 데이터를 연결하는 것이다.
③ 계층형 데이터 모델은 자기 자신에게 조인할 수 없다.
④ 상호배타적 관계에서는 조인키가 배타적으로 상속된다.

**문제 3. 다음 중 트랜잭션에 대한 설명으로 가장 부적절한 것을 고르시오.**
① 트랜잭션은 업무의 논리적 단위이다.
② 트랜잭션의 원자성을 보장받기 위해서는 하나의 커밋 단위로 묶여야 한다. 
③ 트랜잭션의 범위는 모델로는 표현되지 않는다.
④ 잘못된 트랜잭션의 처리는 데이터 품질을 저해한다.

**문제 4. 다음 중 Nul 속성에 대한 설명으로 가장 부적절한 것을 고르시오.**
① Null은 문자의 경우 "와 숫자의 경우 0과 같다.
② Null 값에 대한 사칙 연산의 결과는 언제나 Null이다.
③ 집계함수에서 Null은 제외된다.
④ Null은 is null/is not null 연산만 가능하다.

**문제 5. 다음 중 인조식별자에 대한 설명으로 가장 부적절한 것을 고르시오.**
① 인조식별자는 불필요한 인덱스를 발생시킬 수 있다.
② 인조식별자를 활용하면 본질적인 데이터의 중복을 차단할 수 있다.
③ 인조식별자는 식별자의 수가 많아졌을 경우 고려하는 것이 바람직하다. 
④ 인조식별자는 반드시 본질식별자와의 장단점을 고려하여 생성해야 한다.

**Comment by white-gyu** [comment, marked as answer]
### 연습문제

**문제 1. 아래 데이터 모델을 보고 몇 정규형 위배인지 고르시오.**

![스크린샷 2023-04-13 오후 9 52 11](https://user-images.githubusercontent.com/67765871/231764241-83589055-db68-4c6e-b5b6-4b551421ff05.png)

① 제1정규형
② 제2정규형
③ 제3정규형
④ 보이스코드정규형

> ②, 지도 교수/학과명이 학번에 종속

**문제 2. 다음 중 조인에 대한 설명으로 가장 부적절한 것을 고르시오.** 
① 조인은 엔터티 간 관계에서 비롯된다.
② 조인은 매핑키를 통해 데이터를 연결하는 것이다.
③ 계층형 데이터 모델은 자기 자신에게 조인할 수 없다.
④ 상호배타적 관계에서는 조인키가 배타적으로 상속된다.

> ③, 자기 자신에게 join

**문제 3. 다음 중 트랜잭션에 대한 설명으로 가장 부적절한 것을 고르시오.**
① 트랜잭션은 업무의 논리적 단위이다.
② 트랜잭션의 원자성을 보장받기 위해서는 하나의 커밋 단위로 묶여야 한다. 
③ 트랜잭션의 범위는 모델로는 표현되지 않는다.
④ 잘못된 트랜잭션의 처리는 데이터 품질을 저해한다.

> ③

**문제 4. 다음 중 Nul 속성에 대한 설명으로 가장 부적절한 것을 고르시오.**
① Null은 문자의 경우 "와 숫자의 경우 0과 같다.
② Null 값에 대한 사칙 연산의 결과는 언제나 Null이다.
③ 집계함수에서 Null은 제외된다.
④ Null은 is null/is not null 연산만 가능하다.

> ①

**문제 5. 다음 중 인조식별자에 대한 설명으로 가장 부적절한 것을 고르시오.**
① 인조식별자는 불필요한 인덱스를 발생시킬 수 있다.
② 인조식별자를 활용하면 본질적인 데이터의 중복을 차단할 수 있다.
③ 인조식별자는 식별자의 수가 많아졌을 경우 고려하는 것이 바람직하다. 
④ 인조식별자는 반드시 본질식별자와의 장단점을 고려하여 생성해야 한다.

> ②, id만 다르고 나머지 정보 동일한 경우(법인 회원 가입 시 대리인 없는 경우)

**Comment by beolggoolosori** [comment]
문제 1. 아래 데이터 모델을 보고 몇 정규형 위배인지 고르시오.
② 제2정규형

문제 2. 다음 중 조인에 대한 설명으로 가장 부적절한 것을 고르시오.
③ 계층형 데이터 모델은 자기 자신에게 조인할 수 없다.

문제 3. 다음 중 트랜잭션에 대한 설명으로 가장 부적절한 것을 고르시오.
① 트랜잭션은 업무의 논리적 단위이다.

문제 4. 다음 중 Nul 속성에 대한 설명으로 가장 부적절한 것을 고르시오.
① Null은 문자의 경우 "와 숫자의 경우 0과 같다.

문제 5. 다음 중 인조식별자에 대한 설명으로 가장 부적절한 것을 고르시오.
② 인조식별자를 활용하면 본질적인 데이터의 중복을 차단할 수 있다.
